### PR TITLE
Add timeout for sync link

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: src/containers/Sync/Sync.jsx [hunk 1]
```diff
@@ -91,7 +94,9 @@ class Sync extends Component {
     if (this.props.syncStep === 'finished') {
       return this.props.onClose()
     }
-
+    if (this.props.syncStep === 'error') {
```
Comment (G-Ray) [inline src/containers/Sync/Sync.jsx:102]: I think we should let the modal opened in order to display the error to the user ?

File: src/lib/sync-profile.js [hunk 2]
```diff
@@ -77,6 +77,10 @@ class SyncProfile {
     this.peer = peer
   }
 
+  getSw () {
```
Comment (G-Ray) [inline src/lib/sync-profile.js]: no need to set a getSw method, you can already directly access this.sw

File: src/containers/Sync/Sync.jsx [hunk 3]
```diff
@@ -50,18 +51,20 @@ class Sync extends Component {
 
       if (msg !== 'pullProfile') throw new Error('Unexpected message')
       await this.sp.init(channel, Buffer.from(key, 'base64'))
-      await this.sp.joinSecureChannel()
+      await promiseTimeout(15000, this.sp.joinSecureChannel())
```
Comment (G-Ray) [inline src/containers/Sync/Sync.jsx:59]: good

File: src/modals/QRCodeModal/QRCodeModal.jsx [hunk 4]
```diff
@@ -25,11 +25,13 @@ const QRCodeModal = ({ onClose, profile }) => {
   const [link, setLink] = useState('')
   const [syncStep, setSyncStep] = useState(null)
   const [message, setMessage] = useState(null)
+  const [sp, setSp] = useState(null)
 
   useEffect(() => {
     const startSync = async () => {
       const sp = new SyncProfile()
       await sp.init()
+      setSp(sp)
```
Comment (G-Ray) [inline src/modals/QRCodeModal/QRCodeModal.jsx]: why using a state for sp ?

File: src/modals/QRCodeModal/QRCodeModal.jsx [hunk 5]
```diff
@@ -74,12 +76,18 @@ const QRCodeModal = ({ onClose, profile }) => {
     setTimeout(() => setCopied(false), 3000)
   }
 
+  const handleClose = () => {
+    const sw = sp.getSw()
+    sw.close()
```
Comment (G-Ray) [inline src/modals/QRCodeModal/QRCodeModal.jsx]: you can directly call this.sp.sw.close()

File: src/lib/sync-profile.js (hunk 2 above)
Comment (leventdem) [reply src/lib/sync-profile.js]: yes, that's right.

File: src/containers/Sync/Sync.jsx (hunk 1 above)
Comment (leventdem) [reply src/containers/Sync/Sync.jsx:102]: The modal is opened, the purpose of the if statement is only to not execute the code above.

File: src/modals/QRCodeModal/QRCodeModal.jsx (hunk 4 above)
Comment (leventdem) [reply src/modals/QRCodeModal/QRCodeModal.jsx]: In order to get the same instance of SyncProfile instantiated in startSync.

File: src/modals/QRCodeModal/QRCodeModal.jsx (hunk 5 above)
Comment (leventdem) [reply src/modals/QRCodeModal/QRCodeModal.jsx]: this.sp is not accessible here, we are outside of startSync method

**Review by G-Ray — APPROVED** [review]

**Pull request closed without merging**